File: FreshFood/freshfood_app/lib/models/user_address.dart
class UserAddress {
  final int userAddressId;
  final String? label;
  final String recipientName;
  final String? phone;
  final String addressLine;
  final bool isDefault;
  final String? createdAt;

  const UserAddress({
    required this.userAddressId,
    required this.label,
    required this.recipientName,
    required this.phone,
    required this.addressLine,
    required this.isDefault,
    required this.createdAt,
  });

  factory UserAddress.fromJson(Map<String, dynamic> json) {
    final idRaw = json['userAddressID'] ?? json['UserAddressID'] ?? json['userAddressId'] ?? json['userAddressID'];
    final id = idRaw is num ? idRaw.toInt() : int.tryParse('$idRaw') ?? 0;
    return UserAddress(
      userAddressId: id,
      label: json['label']?.toString() ?? json['Label']?.toString(),
      recipientName: (json['recipientName'] ?? json['RecipientName'] ?? '').toString(),
      phone: json['phone']?.toString() ?? json['Phone']?.toString(),
      addressLine: (json['addressLine'] ?? json['AddressLine'] ?? '').toString(),
      isDefault: (json['isDefault'] ?? json['IsDefault']) == true,
      createdAt: json['createdAt']?.toString() ?? json['CreatedAt']?.toString(),
    );
  }
}

File: FreshFood/freshfood-be/freshfood-be/Models/UserAddress.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace freshfood_be.Models;

[Table("UserAddresses")]
public class UserAddress
{
    [Key]
    public int UserAddressID { get; set; }

    public int UserID { get; set; }
    [ForeignKey("UserID")]
    [JsonIgnore]
    public User? User { get; set; }

    [StringLength(60)]
    public string? Label { get; set; }

    [Required]
    [StringLength(100)]
    public string RecipientName { get; set; } = string.Empty;

    [StringLength(20)]
    public string? Phone { get; set; }

    [Required]
    [StringLength(500)]
    public string AddressLine { get; set; } = string.Empty;

    public bool IsDefault { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
